voila/handler.py

```diff
@@ -118,12 +113,29 @@ async def get(self, path=None):
     def redirect_to_file(self, path):
         self.redirect(url_path_join(self.base_url, 'voila', 'files', path))
 
-    @tornado.gen.coroutine
-    def _jinja_kernel_start(self):
+    async def _jinja_kernel_start(self, nb):
         assert not self.kernel_started, "kernel was already started"
-        # See command above aboout not being able to start the kernel from a different thread
+        kernel_id = await self.kernel_manager.start_kernel(kernel_name=self.notebook.metadata.kernelspec.name, path=self.cwd)
+        km = self.kernel_manager.get_kernel(kernel_id)
+        # When Voila is launched as an app, its kernel manager's type is AsyncMappingKernelManager, and thus
+        # its kernel client's type is AsyncKernelClient.
+        # But this has to be explicitly configured when launched as a server extension, with e.g.:
+        # --ServerApp.kernel_manager_class=jupyter_server.services.kernels.kernelmanager.AsyncMappingKernelManager
+        # If it's not done, the kernel manager might not be async, which is not a big deal, but we want the kernel
+        # client to be async, so we explicitly configure it for this particular case:
+        km.client_class = 'jupyter_client.asynchronous.AsyncKernelClient'
+        self.executor = VoilaExecutor(nb, km=km, config=self.traitlet_config)
+        self.executor.kc = km.client()
+        self.executor.kc.start_channels()
```

Comment:
I am/was a bit worried about this. I remember when I just started adding an async client that the jupyter_server part used the same client. Maybe that was indeed because I configured it to be async.
Are you saying that both jupyter_server and voila have their own client to the kernel, jupyter_server does whatever it does, and we do it async?
Did you check if https://github.com/voila-dashboards/voila/blob/master/notebooks/basics.ipynb works with both the voila app and server extension?

Reply:
Yes it works both with the voila app and the server extension. I'm not sure I understand what you mean, Voila has its own kernel anyway, right?

Reply:
Well, the kernel is also known to jupyter server, which is responsible for the WebSocket connection. So jupyter server had a connection to the kernel as well (actually it may be in parallel to us)

Reply:
could we use nbclient's async context manager `async with self.async_setup_kernel()` instead of manually doing

```
        self.executor.kc = km.client()
        self.executor.kc.start_channels()
```